.pre-commit-config.yaml

```diff
@@ -28,6 +28,15 @@ repos:
   hooks:
   - id: black
 
+- repo: https://github.com/pre-commit/mirrors-mypy
+  rev: "v1.1.1"
+  hooks:
+  - id: mypy
+    files: src
+    args: []
+    additional_dependencies: [types-setuptools]
+    stages: [manual]
+
```

Comment:
The addition of mypy is a bit premature for wheel. With the public API work it will become more relevant, but that's not yet the case in `main`.

Reply:
Also, I let the `ruff` change slide in the previous PR, but generally my policy is to separate such changes to separate PRs.

Reply:
It's only a manual hook (as it's not passing yet on the rest of the codebase), but it was used to verify the new code was fine. I can remove the manual check if you'd like, but it can only be triggered with `--hook-stage=manual`, it doesn't run automatically or in CI.

Reply:
Dropped it. Might try to make a PR adding mypy eventually - it it was in it's own PR, I could fix the places it is unhappy with, which I can't do in this PR.

Reply:
Yup, I noticed that. I will add this myself to wheel when I feel the time is right.